composer/utils/checkpoint.py

```diff
@@ -601,6 +602,57 @@ def load_checkpoint(
     return rng_state_dicts
 
 
+def dist_cp_load(
+    state_dict: dict[str, Any],
+    storage_reader: StorageReader,
+    load_planner: Optional[LoadPlanner] = None,
+):
```

Comment:
can we put a docstring on this fn explaining the issue as well? something roughly equivalent to PR description